server/routes/all-episode-data.js

```diff
@@ -3,49 +3,94 @@
-
-// in memory cache of s3/cloudfront versions for each episode JSON file (waveform + transcript)
-let episodeDataVersions = {};
+let cache;
 
 // update `allEpisodes` and `episodeDataVersions`
-const update = function(cb) {
+const update = function(globalCache, cb) {
   cb = cb || function(err, success) {console.log('update show data', err || success);};
+  cache = globalCache;
 
-  // get show data from TAL's API
-  request.get({url: dataUrl, rejectUnauthorized: false}, function(err, res, body) {
-    if (!err) {
-      try {
-        // update the value of allEpisodes
-        allEpisodes = JSON.parse(body).map((item) => {
-          return {
-            'number': item.number,
-            'description': item.description,
-            'original_air_date': item.original_air_date,
-            'title': item.title
-          };
+  // get show data
+  if (rssFeed) {
+    request.get({url: rssFeed}, function(err, resp, body) {
+      if (!err) {
+        let episodes = cache.getKey('episodes') || [];
+        console.log('AAAA',episodes);
+        helpers.parseRSS(body, episodes, function(result) {
+          if (result.err) {
+            return cb(result.err);
+          }
+          else {
+            cache.setKey('allEpisodes', result.episodes);
+            cache.setKey('allEpisodesUnfiltered', result.episodesUnfiltered);
+            cache.save(true);
+            return cb(null, 'success');
+          }
         });
-        return cb(null, 'success');
-      } catch(e) {
-        console.error('unable to parse full latest episodes from TAL', e);
       }
-    }
-    return cb('no body');
-  });
+      else {
+        return cb(err);
+      }
+    });
+  }
+  else {
+    request.get({url: dataUrl, rejectUnauthorized: false}, function(err, res, body) {
+      if (!err) {
+        try {
+          let activeEpisodes = [];
+          let episodes = cache.getKey('episodes');
+          if (Array.isArray(episodes)) {
+            activeEpisodes = episodes
+                              .filter(episode => episode.enabled === true)
+                              .map(episode => episode.value);
+          }
+          // update the value of allEpisodes
+          let unfilteredEpisodes = JSON.parse(body).map((item) => {
+            return {
+              'number': item.number,
+              'description': item.description,
+              'original_air_date': item.original_air_date,
+              'title': item.title,
+              'guid': item.number
+            };
+          });
+          cache.setKey('allEpisodesUnfiltered', unfilteredEpisodes);
+          let filteredEpisodes = unfilteredEpisodes
+            // filter out inactive episodes specified in the .env file
+            .filter((episode) => !inactiveEpisodes.includes(episode.number))
+            // filter out inactive episodes (only include explicitly active episodes in the admin panel)
+            // unless we passed `true` to `episodes` in which case, include all episodes
```

Comment:
when do we pass `true` instead of an array of episodes?

Reply:
This stuff was left over from an earlier instance of the code, before I gave the option to return filtered or unfiltered episodes. Fixing now!